geopandas/io/file.py

```diff
@@ -414,16 +421,27 @@ def infer_schema(df):
     from collections import OrderedDict
 
     # TODO: test pandas string type and boolean type once released
-    types = {"Int64": "int", "string": "str", "boolean": "bool"}
+    types = {
+        "int64": "int64",
+        "int32": "int32",
+        "int16": "int",
+        "int8": "int",
```

Comment:
Just noting that mapping `int32` to `int32` is a behaviour change (I think a good one). Under the old resolution
`type(np.zeros(1, np.dtype('int32').item()).__name__` returns `"int"` which would then become an `int64` when written to file.
```suggestion
        "int32": "int32",
        "int16": "int32",
        "int8": "int32",
```
I think it makes sense to write 16 and 8 bit ints as 32 bit rather than 64 bit (see also that same comment above from the old PR)